tests/r8.rs

```diff
@@ -98,3 +99,42 @@ fn test_uuid_win() {
         "d8b03b44-58df-5cd7-adc7-aefcfb0e2ade".parse().unwrap()
     );
 }
+
+#[test]
+fn test_remap_source_file() {
+    let mapping = ProguardMapping::new(MAPPING_R8_SYMBOLICATED_FILE_NAMES);
+
+    let mapper = ProguardMapper::new(mapping);
+
+    let test = mapper.remap_stacktrace(r#"
+    Caused by: java.lang.Exception: Hello from main!
+	at a.a.a(SourceFile:12)
+	at io.wzieba.r8fullmoderenamessources.MainActivity.b(SourceFile:6)
+	at io.wzieba.r8fullmoderenamessources.MainActivity.a(SourceFile:1)
+	at a.c.onClick(SourceFile:1)
+	at android.view.View.performClick(View.java:7659)
+	at android.view.View.performClickInternal(View.java:7636)
+	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)"#
+    );
+
+    /**
+    Ideally, should be (output from `retrace` tool):
+    Caused by: java.lang.Exception: Hello from main!
+   	at io.wzieba.r8fullmoderenamessources.Foobar.foo(Foobar.java:10)
+   	at io.wzieba.r8fullmoderenamessources.MainActivity.onCreate$lambda$1$lambda$0(MainActivity.java:14)
+   	at io.wzieba.r8fullmoderenamessources.MainActivity.$r8$lambda$pOQDVg57r6gG0-DzwbGf17BfNbs(MainActivity.java:0)
+   	at io.wzieba.r8fullmoderenamessources.MainActivity$$ExternalSyntheticLambda0.onClick(MainActivity.java:0)
+   	at android.view.View.performClick(View.java:7659)
+   	at android.view.View.performClickInternal(View.java:7636)
+   	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
+     */
+    assert_eq!(r#"
+    Caused by: java.lang.Exception: Hello from main!
+    at io.wzieba.r8fullmoderenamessources.Foobar.foo(Foobar.kt:10)
+    at io.wzieba.r8fullmoderenamessources.MainActivity.onCreate$lambda$1$lambda$0(MainActivity.kt:14)
+    at io.wzieba.r8fullmoderenamessources.MainActivity.$r8$lambda$pOQDVg57r6gG0-DzwbGf17BfNbs(MainActivity.kt:0)
+    at io.wzieba.r8fullmoderenamessources.MainActivity$$ExternalSyntheticLambda0.onClick(R8$$SyntheticClass:0)
```

Comment:
`R8$$SyntheticClass` here and in other places should be replaced with the correct class name, e.g. `MainActivity`. I haven't yet explored how to do this though (see `retrace` output in the comment above).

Reply:
Yes, I think you're right. I've checked their docs, but they only mention what's `# {'id':'com.android.tools.r8.synthesized'}` - nothing about `R8$$SyntheticClass`-likes.

Also, looking at the mappings and stack trace, I don't see other rule than suggested by you substring of the last part of the classname 👍. I'll try to come up with logic to cover this case

Reply:
> substrings the last part of the classname for synthetic classes

But we don't have the information about file extension then. I can see two solutions:

1. Treat all files are `.java` (this is what `retrace` seems to be doing: https://github.com/getsentry/rust-proguard/pull/36#discussion_r1547749866)
2. Take the last part of the classname and then look for `{"id": "sourceFile" ...}` declarations if we have a matching file - if yes, then get the file name + extension from there.

I like the idea nr 1 more as it's simpler and in sync with `retrace`. But I'm worried that it might conflict with some other Sentry features in the dashboard, like Source Context. WDYT?

Reply:
source context should be fine as we disregard the file extension there. For me it'd be even fine to leave the filename without any extension at all tbh, but 1. sounds good as well.

Reply:
> source context should be fine as we disregard the file extension there

Gotcha, good to know 👍 

>  For me it'd be even fine to leave the filename without any extension at all tbh, but 1. sounds good as well.

To clarify, because I see I wrote my suggestion in a confusing way: I wanted to suggest:
- keeping file extension if we know the file (from R8 clousure, like `#" {"id":"sourceFile","fileName":"...`)
- adding `.java` extension only to files we take from the class name (the case of `R8$$SyntheticClass`)

I hope that's cool. This PR introduces the first point, I have an issue with adding `.java` suffix though https://github.com/getsentry/rust-proguard/pull/36#discussion_r1548255732

Reply:
yeah, I meant for the latter part

> adding .java extension only to files we take from the class name (the case of R8$$SyntheticClass)

if we can't infer the file extension, in this case, we could as well just skip it altogether. But I'm also fine with adding `.java`